upup/pkg/fi/nodeup/nodetasks/service.go

```diff
@@ -67,11 +72,22 @@ func (p *Service) GetDependencies(tasks map[string]fi.Task) []fi.Task {
 		// LoadImageTask or IssueCert. If there are any LoadImageTasks (e.g. we're
 		// launching a custom Kubernetes build), they all depend on
 		// the "docker.service" Service task.
-		switch v.(type) {
-		case *File, *Package, *UpdatePackages, *UserTask, *GroupTask, *Chattr, *BindMount, *Archive:
+		switch v := v.(type) {
+		case *Package, *UpdatePackages, *UserTask, *GroupTask, *Chattr, *BindMount, *Archive:
 			deps = append(deps, v)
-		case *Service, *LoadImageTask, *IssueCert:
+		case *Service, *LoadImageTask, *IssueCert, *BootstrapClient:
 			// ignore
+		case *File:
+			// Skip adding kubeconfig files as dependencies for services other than kubelet
+			if strings.HasSuffix(v.Path, "/kubeconfig") && p.Name != kubeletService {
```

Comment:
I have a preference for something more explicit and closer to the Builders than looking at the path suffix. One idea would be to add a `BeforeService string` field to `File` and populate it for all the kubeconfigs that can derive from `BootstrapClient`. The value `""` would make all `Service`s depend on it.

Reply:
I think the latest changes should address this.